README.md

```diff
@@ -1,3 +1,161 @@
+pip install fairgbm/python-package
+```
+
+## Examples
+
+A short Python code snippet:
+```python
+from fairgbm import FairGBMClassifier
+
+# Instantiate
+fairgbm_clf = FairGBMClassifier(
+    "constraint_type"="FNR",    # constraint on equal group-wise TPR (equal opportunity)
+    "n_estimators"=200,         # core parameters from vanilla LightGBM
+    "random_state"=42,          # ...
+)
+
+# Train using features (X), labels (Y), and sensitive attributes (S)
+fairgbm_clf.fit(X, Y, constraint_group=S)
+
+# Predict
+Y_test_pred = fairgbm_clf.predict_proba(X_test)[:, -1]  # Compute continuous class probabilities (recommended)
+# Y_test_pred = fairgbm_clf.predict(X_test)             # Or compute discrete class predictions
+```
+
+A more in-depth explanation and other usage examples can be found in the [**_examples folder_**](/examples).
+
+For Python examples see the [**_notebooks folder_**](/examples/FairGBM-python-notebooks).
+
+
+## Features
+
+FairGBM enables you to train a GBM model to **minimize a loss function** (_e.g._, cross-entropy) **subject to fairness 
+constraints** (_e.g._, equal opportunity).
+
+Namely, you can target equality of performance metrics (FPR, FNR, or both) across instances from different protected groups (see [fairness constraints](#fairness-constraints) section).
+Simultaneously (and optionally), you can add global constraints on specific metrics (see [global constraints](#global-constraints) section).
+
+### Parameter list
+
+List of FairGBM-specific parameters:
+**TODO**: enumerate parameters and document each one;
+
+| _Name_ | _Description_ | _Default_ |
+|:------:|---------------|:---------:|
+| `groupwise_constraint_type` | The type of fairness (group-wise equality) constraint to use (if any). | `FPR,FNR` |
+| `global_constraint_type` | The type of global equality constraint to use (if any). | _None_ |
+| ... | **TODO** |
+
+Besides these FairGBM-specific parameters, you can use any core LightGBM parameter as well (_e.g._, 
+`n_estimators`). The only exception is the `objective` parameter, as FairGBM only uses 
+`objective="constrained_cross_entropy"` (a standard non-constrained objective will fallback to using standard LightGBM).
+
+Please consult [this list](https://lightgbm.readthedocs.io/en/latest/Parameters.html#core-parameters) for a detailed 
+view of all vanilla LightGBM parameters.
+
+
+### Fairness constraints
+
+You can use FairGBM to equalize the following metrics across protected groups:
+- Equalize FNR (equivalent to equalizing TPR or Recall)
+    - also known as _equal opportunity_ [(Hardt et al., 2016)](#external-references)
+- Equalize FPR (equivalent to equalizing TNR or Specificity)
+    - also known as _predictive equality_ [(Corbett-Davies et al., 2017)](#external-references)
+- Equalize both FNR and FPR simultaneously
+    - also known as _equal odds_ [(Hardt et al., 2016)](#external-references)
+
+<!--
+Take the following hypothetical example:
```

Comment:
Will we use this example?

Reply:
That's what I'm trying to figure out. I wrote it and left it for PR reviewers, I think it is useful information but we should strive to keep the main README as short as possible; perhaps this should be somewhere else or we should just point people to the paper